src/Symfony/Component/Security/Http/Authenticator/Passport/Passport.php

```diff
@@ -31,13 +32,24 @@ class Passport implements UserPassportInterface
     private $attributes = [];
 
     /**
-     * @param CredentialsInterface $credentials the credentials to check for this authentication, use
-     *                                          SelfValidatingPassport if no credentials should be checked
+     * @param string|UserBadge     $usernameOrBadge a username to use for the user loader or a UserBadge instance
+     * @param CredentialsInterface $credentials     the credentials to check for this authentication, use
+     *                                              SelfValidatingPassport if no credentials should be checked
      * @param BadgeInterface[]     $badges
      */
-    public function __construct(UserInterface $user, CredentialsInterface $credentials, array $badges = [])
+    public function __construct($usernameOrBadge, CredentialsInterface $credentials, array $badges = [])
     {
-        $this->user = $user;
+        if ($usernameOrBadge instanceof UserInterface) {
+            trigger_deprecation('symfony/security-http', '5.2', 'The user argument of "%s" must be a "%s" or a string containing the username.', __CLASS__, UserBadge::class);
+
+            $this->user = $usernameOrBadge;
+        } elseif ($usernameOrBadge instanceof UserBadge) {
+            $this->addBadge($usernameOrBadge);
+        } elseif (\is_string($usernameOrBadge)) {
```

Comment:
I'm wondering if we should only allow `UserBadge` instances. Using `new UserBadge($username)` is a bit more explicit than `$username` and still quite simple to understand IMHO.